src/components/inputs/Select.test.tsx

```diff
@@ -0,0 +1,328 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React from 'react';
+
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { forEach, map, reject, slice } from 'lodash';
+
+import { render } from '../../test-utils';
+import { Select } from './Select';
+
+const items = [
+	{
+		label: 'hi',
+		value: '1'
+	},
+	{
+		label: 'hello',
+		value: '2'
+	},
+	{
+		label: 'good day',
+		value: '3'
+	},
+	{
+		label: 'goodnight',
+		value: '4'
+	},
+	{
+		label: 'nothing',
+		value: '5'
+	}
+];
+
+const extendedItems = [
+	...items,
+	{
+		label: 'unselectable',
+		value: '6',
+		disabled: true
+	}
+];
+
+describe('Select', () => {
+	const label = 'Select an item';
+	const onChange = jest.fn();
+	describe('single selection ', () => {
+		test('disabled items are not selectable and does not trigger onChange', async () => {
+			render(
+				<Select items={extendedItems} label={label} onChange={onChange} selection={items[0]} />
+			);
+
+			await userEvent.click(screen.getByText(label));
+			await userEvent.click(screen.getByTestId(extendedItems[5].label));
+
+			expect(onChange).not.toBeCalled();
+			expect(screen.getByTestId(extendedItems[0].label)).toBeVisible();
```

Comment:
Yes you are correct. There are two string with the same text, one for the label of the select and one for the label of the dropdown option. I wanted to make sure to avoid conflicts between those two adding a test id to the dropdown options but I am open to cleaner alternatives.

In this case I want the label of the select to be the same as the label of the dropdown option

Reply:
So, in order to follow what the user would see, we can check that there is only 1 label for the disabled option and 2 (and 2 only) labels visible for the selected option. Transformed in code:
```ts
expect(screen.getByText(extendedItems[5].label)).toBeVisible();
const selectedItemLabels = screen.getAllByText(extendedItems[0].label);
expect(selectedItemLabels).toHaveLength(2);
expect(selectedItemLabels[0]).toBeVisible();
expect(selectedItemLabels[1]).toBeVisible();
```